### DISCUSSION TITLE
Read only new emails since last execution?

### DISCUSSION THREAD

**Discussion by dankykang331** [umanamente/n8n-nodes-imap#27] [Q&A] [answered]
I have trouble with the default IMAP module in 8n, so I'm trying to use this custom module to read emails every five minutes. Is there a way to set it up so that I will only retrieve new emails since the last time the workflow was executed?

**Comment by umanamente** [comment, marked as answer]
There is a 'Date Range' filter. You could store the last execution time and pass it to the 'Since' field.

**Reply by dankykang331** [reply]
Perfect, thanks.

**Comment by webkp** [comment]
@dankykang331 I had problems with the default IMAP node too but to avoid a schedule every x minutes I still use the default node but only as trigger on new emails and offload all other work to @umanamente's node - works great and runs only on new mails. I thought I'd mention it in case it helps anyone.

@umanamente thank you a lot for your great node! 🥳
